### PR TITLE
Start implementing shaders-based heatmap behind flag

### PR DESCRIPTION
**Pull request by axelboc**
This is an initial implementation to get things moving. For now, I use the inferno colour map function in the shader (instead of passing the colour map as a texture, as discussed). Also, only the linear scale is supported.

image

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -18,6 +18,9 @@ REACT_APP_HSDS_FALLBACK_DOMAIN="/home/reader/water"
 REACT_APP_JLAB_URL=
 REACT_APP_JLAB_FALLBACK_DOMAIN="water_224.h5"
 
+# Development-only feature flags
+REACT_APP_DEV_SHADERS=false
+
```
Comment (axelboc) [inline .env:23]: Just turn it on in `.env.local` to switch to the new shaders-based mesh. 😉

File: src/h5web/vis-packs/core/heatmap/Mesh.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ function Mesh(props: Props): ReactElement {
     <>
       {material && (
         <mesh material={material}>
-          <planeBufferGeometry attach="geometry" args={[width, height]} />
+          <planeGeometry attach="geometry" args={[width, height]} />
```
Comment (axelboc) [inline src/h5web/vis-packs/core/heatmap/Mesh.tsx:48]: Turns out that `PlaceBufferGeometry` is just an alias. `PlaneGeometry` is already a `BufferGeometry`.

File: src/h5web/vis-packs/core/heatmap/ShadersMesh.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  const shader = useMemo(
+    () => ({
+      uniforms: {
+        data: {
+          value: new DataTexture(
+            Float32Array.from(values),
+            cols,
+            rows,
+            RedFormat,
```
Comment (axelboc) [inline src/h5web/vis-packs/core/heatmap/ShadersMesh.tsx:24]: `RedFormat` avoids having to add three zeros for each value to our `values` array before passing it to the `DataTexture`.

File: src/h5web/vis-packs/core/heatmap/ShadersMesh.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <mesh scale={[width, height, 1]}>
+      <planeGeometry attach="geometry" args={[1, 1]} />
+      <shaderMaterial attach="material" args={[shader]} />
+    </mesh>
+  );
+}
+
+export default memo(ShadersMesh);
```
Comment (loichuder) [inline src/h5web/vis-packs/core/heatmap/ShadersMesh.tsx:79]: If the `ShadersMesh` is memoized, it has the props as deps, right ? Meaning `[rows, cols, values, domain]`.

In this case, is the `useMemo` with the same deps really useful ?

**Review by loichuder — APPROVED** [review]
It is so faaaaaaaaast :rocket:

File: src/h5web/vis-packs/core/heatmap/ShadersMesh.tsx (hunk 4 above)
Comment (axelboc) [reply src/h5web/vis-packs/core/heatmap/ShadersMesh.tsx:79]: Excellent point!

**Pull request merged**